### PR TITLE
Python binding overhaul, automatic binary and documentation generation

### PR DESCRIPTION
**Pull request by jackgerrits**
- Replaces SWIG bindings with PyBind11
- Create Linux CI that will produce installable binary wheels for Linux Python 3.6, 3.7, 3.8, 3.9
    - Artifacts you can try here: https://github.com/jackgerrits/reinforcement_learning/actions/runs/861950993
- Proper Python generated documentation
- Migration guide for these new bindings

See here for CI run: https://github.com/jackgerrits/reinforcement_learning/runs/2634424830

Docs preview:
[docs.zip](https://github.com/VowpalWabbit/reinforcement_learning/files/6518897/docs.zip)

### CODE REVIEW

File: examples/python/rl_sim.py [hunk 1]
```diff
@@ -37,8 +37,7 @@ def __init__(self, args):
         self._options = args
 
         self.config = load_config_from_json(self._options.json_config)
-        self._rl_client = rl_client.live_model(self.config, my_error_callback())
-        self._rl_client.init()
+        self._rl_client = rl_client.live_model(self.config, on_error)
```
Comment (zwd-ms) [inline examples/python/rl_sim.py:40]: LiveModel

File: examples/python/basic_usage.py [hunk 2]
```diff
@@ -1,45 +1,43 @@
...
 def load_config_from_json(file_name):
     with open(file_name, 'r') as config_file:
         return rl_client.create_config_from_json(config_file.read())
 
 def main():
     config = load_config_from_json("client.json")
-
-    test_cb = my_error_callback()
-    model = rl_client.live_model(config, test_cb)
+    model = rl_client.live_model(config, on_error)
```
Comment (zwd-ms) [inline examples/python/basic_usage.py:16]: LiveModel, remove init

**Review by zwd-ms — APPROVED** [review]
LGTM, nice work

**Review by lalo — APPROVED** [review]

**Comment by zwd-ms** [comment]
(1) as discussed offline, maybe add a cmake toolchain option
(2) update bindings/python/README.md

**Review by lokitoth — APPROVED** [review]

**Pull request merged**